wgsl/index.bs

```diff
@@ -4557,11 +4555,282 @@ TODO: *Stub*: Expression evaluation
+
+<table class='data'>
+  <caption>Uniformity rules for statements</caption>
+  <thead>
+    <tr><th>Statement<th>New vertices<th>Recursive analyses<th>Behaviors<th>Resulting control-flow vertex<th>New edges
+  </thead>
+  <tr><td class="nowrap">*s1*; *s2*
+      <td>
+      <td class = "nowrap">*CF*, *s1* -> *CF1*, *B1*<br>
+          *CF1*, *s2* -> *CF2*, *B2*
+      <td>*B1* \ {Nothing} U *B2*
+      <td>*CF2*
+      <td>
+  <tr><td class="nowrap">if (*e*) then *s1* else *s2*
+      <td>*CFinside*, *CFend*
+      <td class="nowrap">*CF*, *e* -> *CF'*, *V*, *B*<br>
+          *CFinside*, *s1* -> *CF1*, *B1*<br>
+          *CFinside*, *s2* -> *CF2*, *B2*
+      <td>*B* U *B1* U *B2*
+      <td>*CFend*
+      <td class="nowrap">*CFinside* -> {*CF'*, *V*}<br>
+          *CFend* -> {*CF1*, *CF2*}
+  <tr><td class="nowrap">loop {*s1* continuing {*s2*}}
+      <td>*CF'*
+      <td class="nowrap">*CF'*, *s1* -> *CF1*, *B1*<br>
+          *CF1*, *s2* -> *CF2*, *B2*
+      <td>*B1* \ {Break, Continue} U *B2* U {Nothing}
+      <td>*CF'*
+      <td class="nowrap">*CF'* -> *CF2*
+  <tr><td class="nowrap">loop {*s1*}
+      <td>*CF'*
+      <td class="nowrap">*CF'*, *s1* -> *CF1*, *B1*
+      <td>*B1* \ {Break, Continue} U {Nothing}
+      <td>*CF'*
+      <td class="nowrap">*CF'* -> *CF1*
+  <tr><td class="nowrap">break;
+      <td>
+      <td>
+      <td>{Break}
+      <td>*CF*
+      <td>
+  <tr><td class="nowrap">continue;
+      <td>
+      <td>
+      <td>{Continue}
+      <td>*CF*
+      <td>
+  <tr><td class="nowrap">discard;
+      <td>
+      <td>
+      <td>{Discard}
+      <td>*CF*
+      <td>
+  <tr><td class="nowrap">return;
+      <td>
+      <td>
+      <td>{Return}
+      <td>*CF*
+      <td>*CF_return* -> *CF*
+  <tr><td class="nowrap">return *e*;
+      <td>
+      <td class="nowrap">*CF*, *e* -> *CF'*, *V*, *B*
+      <td>B \ {Nothing} U {Return}
+      <td>*CF'*
+      <td>*CF_return* -> *CF'*<br>
+          *Value_return* -> *V*
+  <tr><td class="nowrap">e1 = e2;
+      <td>
+      <td class="nowrap">*CF*, *e1* => *CF1*, *L1*, *B1*<br>
+          *CF1*, *e2* -> *CF2*, *V2*, *B2*
+      <td>B1 U B2
+      <td>*CF2*
+      <td>*L1* -> {*V2*, *CF*}
+</table>
+
+TODO: double check the rule for loop. I had verified the rules for "for" and "while", but I have some difficulty wrapping my brain around the "continuing" block.
+
+### Uniformity rules for function calls ### {#uniformity-function-calls}
+
+The most complex rule is for function calls:
+- For each argument, apply the corresponding expression rule, with the control-flow at the exit of the previous argument (using the control-flow at the beginning of the function call for the first argument). Name the corresponding value vertices "arg_i" and the corresponding control-flow vertices "CF_i"
```

Comment:
what are the "CF_i" exactly?

Reply:
They're the nodes returned by the recursive calls to the analysis when analyzing the arguments.
They're intended to correspond to the uniformity requirement of the control-flow at the end of evaluating the corresponding argument.